Tooling cleanup

ahmedelgabri opened the pull request:
- remove ranges from `package.json`
- `npm` scripts [adds executables form the `node_modules` folder in `$PATH` automatically](https://docs.npmjs.com/misc/scripts#path) so you can just reference the executable only without the path
- fix `.stylelintrc` indentation
- cleanup ESLint errors in `webpack/` folder

mmahalwy commented:
Problem coming from `phantomjs`

Would you mind updating our phantomjs package?

ahmedelgabri commented:
@mmahalwy That's not `phamtomjs` problem it's a `node-sass` issue. Updated it.

Also, updated how Jenkins pickup the right node version since it was using node `5`

ahmedre commented:
jenkins does use 6.3 (it uses the same configuration as the `Dockerfile`).

complaint from Jenkins is:

```
> quran@1.0.0 build:client /quran
> node webpack --config webpack/prod.config.js

module.js:442
    throw err;
    ^

Error: Cannot find module '/quran/webpack'
    at Function.Module._resolveFilename (module.js:440:15)
    at Function.Module._load (module.js:388:25)
    at Module.runMain (module.js:575:10)
    at run (bootstrap_node.js:352:7)
    at startup (bootstrap_node.js:144:9)
    at bootstrap_node.js:467:3
```

ahmedelgabri commented:
Thanks for posting this because I can't see the errors myself, will have a look at this tomorrow.

ahmedre commented on .travis.yml:
```diff
@@ -1,8 +1,6 @@
 language: node_js
 env: TEST_SUITE=unit
-node_js:
-  - "5.6.0"
```
i prefer putting 6.3.0 here instead of `.nvmrc` just because it will save a lot of pain in the future (next time Travis breaks, people will immediately open `.travis.yml`, and it may not be obvious to someone else that they need to look at `.nvmrc`).

ahmedelgabri replied:
We can add a comment too saying that the node version is handled in `.nvmrc`. 

Also most of the time travis breaks it will be something else plus travis already shows the version in the log & also that it picks it up from `nvm`. 

I'm planning to make the tools & build process more robust so that things doesn't break unless it's something major.

ahmedre commented on package.json:
```diff
@@ -6,165 +6,165 @@
+    "morgan": "1.6.1",
+    "node-sass": "3.8.0",
+    "normalizr": "2.0.0",
+    "phantomjs": "1.9.18",
```
probably to fix the build, set to 1.9.20 and remove it from dev dependencies since it's needed for both

ahmedelgabri replied:
You mean for `phantomjs`?

ahmedre replied:
yep

ahmedelgabri replied:
I think you commented on an old commit of `package.json` because it's already `1.9.20` in this branch.

ahmedelgabri replied:
scratch that, `phantomjs` was in `pakcage.json` twice!

ahmedre commented:
Deployed to: http://staging.quran.com:32839

ahmedre commented:
Deployed to: http://staging.quran.com:32840

ahmedre commented:
Deployed to: http://staging.quran.com:32841

sabeurthabti commented:
Looks good to me. let me know if this is ready to be merged.

mmahalwy commented on Dockerfile:
```diff
@@ -18,7 +18,7 @@ RUN cp /etc/cron.daily/logrotate /etc/cron.hourly
 WORKDIR /tmp
 ADD package.json package.json
-RUN npm install
+RUN npm install --silent --no-progress
```
technically, we should eventually add `--production` to it. So we aren't downloading unneeded dependencies like phantomjs

ahmedelgabri replied:
True, but then travis will fail because the karma test that runs on travis requires `phantomjs`

ahmedre commented:
Deployed to: http://staging.quran.com:32845

ahmedelgabri commented:
@sabeurthabti It's ok form my side too, but to be sure I will just do one last check today afternoon.

ahmedre commented:
Deployed to: http://staging.quran.com:32846

ahmedelgabri commented:
@sabeurthabti You can merge this now. I'll follow up with a new PR to do more cleanup, but for now this is good to go 👍

sabeurthabti commented:
lgtm

sabeurthabti commented:
Thank you @ahmedelgabri

mmahalwy commented:
Thanks @ahmedelgabri !
